src/H5Tcommit.c

```diff
@@ -148,6 +148,7 @@ H5T__commit_api_common(hid_t loc_id, const char *name, hid_t type_id, hid_t lcpl
     new_obj->connector = (*vol_obj_ptr)->connector;
     new_obj->connector->nrefs++;
     new_obj->data = data;
+    new_obj->rc = 1;
```

Comment:
Same with this piece of code (which I admit wasn't originally written by you) - could you please add a small routine in the H5VL code that encapsulated this setup and then call it here (and down below in this routine)?

Reply:
I can replace this with H5VL_create_object, though again there are still other places in the library that peek into H5VL_object_t. In fact this code would still need to do that ((*vol_obj_ptr)->connector), so we would need to write another routine to retrieve that, then propagate it to everywhere that currently gets the connector directly, which includes many (all?) of the *_api_common() routines.

Reply:
Are you thinking we should move H5VL_object_t to H5VLpkg.h? This is kind of outside the scope of this change.

Reply:
Agree - I would volunteer to pick up more of the changes like this in another PR, based on the start you make here.

Reply:
> Are you thinking we should move H5VL_object_t to H5VLpkg.h? This is kind of outside the scope of this change.

I don't think that's necessary (yet?) and agree, would be far outside of the scope for your PR.